src/Credentials/GCECredentials.php

```diff
@@ -126,28 +139,29 @@ public static function onGce(callable $httpHandler = null)
             $httpHandler = HttpHandlerFactory::build();
         }
         $checkUri = 'http://' . self::METADATA_IP;
-        try {
-            // Comment from: oauth2client/client.py
-            //
-            // Note: the explicit `timeout` below is a workaround. The underlying
-            // issue is that resolving an unknown host on some networks will take
-            // 20-30 seconds; making this timeout short fixes the issue, but
-            // could lead to false negatives in the event that we are on GCE, but
-            // the metadata resolution was particularly slow. The latter case is
-            // "unlikely".
-            $resp = $httpHandler(
-                new Request('GET', $checkUri),
-                ['timeout' => 0.3]
-            );
-
-            return $resp->getHeaderLine(self::FLAVOR_HEADER) == 'Google';
-        } catch (ClientException $e) {
-            return false;
-        } catch (ServerException $e) {
-            return false;
-        } catch (RequestException $e) {
-            return false;
+        for ($i = 1; $i <= self::MAX_COMPUTE_PING_TRIES; $i++) {
+            try {
+                // Comment from: oauth2client/client.py
+                //
+                // Note: the explicit `timeout` below is a workaround. The underlying
+                // issue is that resolving an unknown host on some networks will take
+                // 20-30 seconds; making this timeout short fixes the issue, but
+                // could lead to false negatives in the event that we are on GCE, but
+                // the metadata resolution was particularly slow. The latter case is
+                // "unlikely".
+                $resp = $httpHandler(
+                    new Request('GET', $checkUri),
+                    ['timeout' => self::COMPUTE_PING_CONNECTION_TIMEOUT_S]
+                );
+
+                return $resp->getHeaderLine(self::FLAVOR_HEADER) == 'Google';
+            } catch (ClientException $e) {
```

Comment:
Could we replace all these catches with a single reference to [`GuzzleHttp\Exception\GuzzleException`](https://github.com/guzzle/guzzle/blob/master/src/Exception/GuzzleException.php)?

```php
catch (GuzzleException $e) {}
```

Reply:
I tried to catch `GuzzleException` or even the `Exception`, but it seems the Exception is not catched and break the test. https://travis-ci.org/google/google-auth-library-php/jobs/394324374

Reply:
If you catch `Exception`, be sure to add a backslash: `\Exception`. Odd that `GuzzleException` didn't work. You added the `use` statement at the top of the class, right?

Reply:
I added the `use GuzzleHttp\Exception\GuzzleException;` and removed 
```
use GuzzleHttp\Exception\ClientException;
use GuzzleHttp\Exception\RequestException;
use GuzzleHttp\Exception\ServerException;
```

Reply:
@jdpedrie Gentle ping. How does it look now?